cmd/kube-scheduler/app/server.go

```diff
@@ -326,3 +335,10 @@ func newHealthzHandler(config *kubeschedulerconfig.KubeSchedulerConfiguration, s
 	}
 	return pathRecorderMux
 }
+
+// WithPlugin creates an Option based on plugin name and factory.
+func WithPlugin(name string, factory framework.PluginFactory) Option {
+	return func(registry framework.Registry) error {
+		return registry.Register(name, factory)
```

Comment:
I wonder if there is any validation that can be done for the registration.

Reply:
What kind of validation?